src/brevitas/core/scaling/runtime.py

```diff
@@ -182,7 +181,6 @@ def __init__(
     @brevitas.jit.script_method
     def forward(self, x):
         out = x * self.affine_weight + self.affine_bias()
-        out = abs_binary_sign_grad(out)
```

Comment:
This operation happens after we have already clamped.

There are two issues now:
- If we _do_ want to constrain the scale to be positive, then we cannot do it anymore, since the sign can be flipped here
- Not strictly related to this change, but if the weight is zero, and we have already clamped, then we lose that clamping factor.

Problem number 2 is fairly easy to solve, but I'm not sure what to do for problem number one.

Reply:
Actually the solution is simple.
We apply abs_binary_sign_grad to self.affine_weight and that's it